framework/src/main/java/org/checkerframework/framework/flow/CFAbstractTransfer.java

```diff
@@ -366,6 +380,75 @@ public S initialStore(UnderlyingAST underlyingAST, List<LocalVariableNode> param
     return store;
   }
 
+  /**
+   * Determines whether a given lambda expression may be leaked outside the method in which it
+   * appears.
+   *
+   * <p>Currently, a lambda is considered leaked unless it is an argument to a method whose
+   * corresponding formal parameter is annotated as @{@link NonLeaked}. The @{@link NonLeaked}
+   * annotation is trusted, not checked.
+   *
+   * <p>For example, given the following code:
+   *
+   * <pre><code>
+   *   void operateOver(Container container) {
+   *      container.forEach(item -&gt; {...});
+   *   }
+   *
+   *   class Container {
+   *     void forEach(@NonLeaked Consumer&lt;T&gt;)
+   *   }
+   * </code></pre>
+   *
+   * The lambda passed to {@code Container.forEach} is not leaked, as the parameter is annotated
+   * with @{@link NonLeaked}.
+   *
+   * @param lambda the lambda
+   * @param aTypeFactory an annotated type factory
+   * @return true if the lambda may be leaked
+   */
+  private boolean doesLambdaLeak(CFGLambda lambda, AnnotatedTypeFactory aTypeFactory) {
+    LambdaExpressionTree lambdaTree = lambda.getLambdaTree();
+    Tree lambdaParent = aTypeFactory.getPath(lambdaTree).getParentPath().getLeaf();
+    if (lambdaParent.getKind() == Tree.Kind.METHOD_INVOCATION) {
+      MethodInvocationTree invok = (MethodInvocationTree) lambdaParent;
+      ExecutableElement methodElt = TreeUtils.elementFromUse(invok);
+      AliasingAnnotatedTypeFactory aliasingAtf =
+          analysis
+              .atypeFactory
+              .getChecker()
+              .getTypeFactoryOfSubcheckerOrNull(AliasingChecker.class);
+      if (aliasingAtf != null) {
+        // Check whether any parameter is NOT annotated with @NonLeaked
```

Comment:
This should determine the index of `lambdaTree` among the actual arguments of `lambdaParent`, and only check that formal parameter's annotation.